src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -262,6 +261,31 @@ public InnReachTransactionDTO cancelPatronHold(UUID transactionId, CancelPatronH
     return transactionMapper.toDTO(transaction);
   }
 
+  @Override
+  public void returnPatronHoldItem(UUID transactionId, UUID servicePointId) {
+    var transaction = fetchTransactionOfType(transactionId, PATRON);
+
+    verifyState(transaction, ITEM_RECEIVED, RECEIVE_UNANNOUNCED);
+
+    log.info("Attempting to return item of Patron Hold transaction {}", transactionId);
+
+    var loanId = transaction.getHold().getFolioLoanId();
+    var requestId = transaction.getHold().getFolioRequestId();
+    if (loanId != null) {
+      var loan = loanService.getById(loanId);
+      if (loanService.isOpen(loan)) {
+        log.info("Checking-in item for transaction associated with an open loan");
+        loanService.checkInItem(transaction, servicePointId);
```

Comment:
Shouldn't IN_TRANSIT message be sent as well?

Reply:
Check-in leads to loan closure. So we will receive a kafka event for the loan where logic for IN_TRANSIT message will be applied